File: UELPMAPI/UELPM.Model/Models/ReportParameter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace UELPM.Model.Models
{
    public class ReportParameter
    {
        public REPORTTYPE ReportType { get; set; }
        public ReportParamData FilterOptions { get; set; }
        public int WorkflowStatusId { get; set; }
        public int TypeId { get; set; }
        public int UserId { get; set; }
        public DateTime CreatedDateFrom { get; set; }
        public DateTime CreatedDateTo { get; set; }
    }
    public class ReportParam
    {
        public int Id { get; set; }
        public string Name { get; set; }
    }
    public class ReportParamData
    {
        public List<ReportParam> Entities { get; set; }
        public List<ReportParam> Departments { get; set; }
        public List<ReportParam> Statuses { get; set; }
        public List<ReportParam> Requesters { get; set; }
        public List<ReportParam> SupplierTypes { get; set; }
    }

    public enum REPORTTYPE
    {
        SUPPLIER,
        PO,
        POITEMS,
        POCMASTER,
        POC,
        APINVOICE,
        APCREDITNOTE,
        COA,
        POPMASTER,
        POPINVOICE,
        ADMINWORKFLOW,
        CASHFLOW
    }
}

File: UELPMAPP/src/app/reports/models/report-parameter.ts
export class ReportParameter {
    ReportType: REPORTTYPE;
    Entities: any;
    Title: string;
    FilterOptions: ReportParamData;
    WorkflowStatusId: number;
    TypeId: number;
    UserId: number;
    CreatedDateFrom: Date;
    CreatedDateTo: Date;
}
export class ReportColumn {
    field: string;
    header: string;
    dataType: DATATYPE;
    width: number;
    alignment: ALIGNMENT;
    hasFooter?: boolean;
}
export class ReportParamConfig {
    Entity: boolean;
    Department: boolean;
    Status: boolean;
    SupplierType: boolean;
    Requester: boolean;
    constructor(entity: boolean, dept: boolean, status: boolean, supplierTypes: boolean, requester: boolean) {
        this.Entity = entity;
        this.Department = dept;
        this.Status = status;
        this.SupplierType = supplierTypes;
        this.Requester = requester;
    }
}
export class ReportParam {
    Id: number;
    Name: string;
}
export class ReportParamData {
    Entities: ReportParam[];
    Departments: ReportParam[];
    Statuses: ReportParam[];
    Requesters: ReportParam[];
}
export enum ALIGNMENT {
    Left,
    Right,
    Center
}
export enum DATATYPE {
    string,
    currency,
    date,
    number
}
export enum REPORTTYPE {
    SUPPLIER,
    PO,
    POITEMS,
    POCMASTER,
    POC,
    APINVOICE,
    APCREDITNOTE,
    COA,
    POPMASTER,
    POPINVOICE,
    ADMINWORKFLOW,
    CASHFLOW
}